executors.go

```diff
@@ -133,6 +185,24 @@ func (c *Connection) Create(model interface{}, excludeColumns ...string) error {
 						continue
 					}
 
+					if localIsEager {
```

Comment:
I'm not sure the `BeforeSetup` part should be outside the if (in non-eager cases, does it make sense to set the ID field from a struct we didn't touch?). If you confirm that, the whole `if` can be moved outside the loop.

Reply:
I'm pretty sure those are how the m2m/belongs_to are getting set properly in the flat case.